acceptance/tests/connect/connect_helper.go

```diff
@@ -97,31 +153,76 @@ func ConnectInjectConnectivityCheck(t *testing.T, ctx environment.TestContext, c
+	// Create a file called "unhealthy" at "/tmp/" so that the readiness probe
+	// of the static-server pod fails.
+	logger.Log(c.T, "testing k8s -> consul health checks sync by making the static-server unhealthy")
+	k8s.RunKubectl(c.T, c.Ctx.KubectlOptions(c.T), "exec", "deploy/"+staticServerName, "--", "touch", "/tmp/unhealthy")
+
+	// The readiness probe should take a moment to be reflected in Consul,
+	// CheckStaticServerConnection will retry until Consul marks the service
+	// instance unavailable for mesh traffic, causing the connection to fail.
+	// We are expecting a "connection reset by peer" error because in a case of
+	// health checks, there will be no healthy proxy host to connect to.
+	// That's why we can't assert that we receive an empty reply from server,
+	// which is the case when a connection is unsuccessful due to intentions in
+	// other tests.
+	logger.Log(c.T, "checking that connection is unsuccessful")
+	if c.Cfg.EnableTransparentProxy {
+		k8s.CheckStaticServerConnectionMultipleFailureMessages(c.T, c.Ctx.KubectlOptions(c.T), staticClientName, false, []string{
+			"curl: (56) Recv failure: Connection reset by peer",
+			"curl: (52) Empty reply from server",
+			"curl: (7) Failed to connect to static-server port 80: Connection refused",
+		}, "", "http://static-server")
 	} else {
-		k8s.CheckStaticServerConnectionMultipleFailureMessages(t, ctx.KubectlOptions(t), staticClientName, false, []string{"curl: (56) Recv failure: Connection reset by peer", "curl: (52) Empty reply from server"}, "", "http://localhost:1234")
+		k8s.CheckStaticServerConnectionMultipleFailureMessages(c.T, c.Ctx.KubectlOptions(c.T), staticClientName, false, []string{
+			"curl: (56) Recv failure: Connection reset by peer",
+			"curl: (52) Empty reply from server",
+		}, "", "http://localhost:1234")
+	}
+
+	// Return the static-server to a "healthy state".
+	k8s.RunKubectl(c.T, c.Ctx.KubectlOptions(c.T), "exec", "deploy/"+staticServerName, "--", "rm", "/tmp/unhealthy")
+}
+
+// helmValues uses the Secure and AutoEncrypt fields to set values for the Helm
+// Chart which are merged with the HelmValues field with the values set by the
+// Secure and AutoEncrypt fields taking precedence.
+func (c *ConnectHelper) helmValues() map[string]string {
+	helmValues := map[string]string{
+		"connectInject.enabled":        "true",
+		"global.tls.enabled":           strconv.FormatBool(c.Secure),
+		"global.tls.enableAutoEncrypt": strconv.FormatBool(c.AutoEncrypt),
+		"global.acls.manageSystemACLs": strconv.FormatBool(c.Secure),
 	}
+
+	helpers.MergeMaps(helmValues, c.HelmValues)
+
+	return helmValues
+}
+
+// deployFixtureCase looks up the fixture case by name in the `../fixtures/cases`
+// directory and deploys it to the Kubernetes cluster.
+func (c *ConnectHelper) deployFixtureCase(name string) {
+	k8s.DeployKustomize(c.T, c.Ctx.KubectlOptions(c.T), c.Cfg.NoCleanupOnFailure, c.Cfg.DebugDirectory, "../fixtures/cases/"+name)
```

Comment:
I think this can be inlined. IMO given all it is doing is deploying Kustomize, it is easier to read as it rather than extracting it into a method.